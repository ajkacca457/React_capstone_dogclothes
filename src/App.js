import {
  BrowserRouter as Router, Route, Switch, Redirect,
} from 'react-router-dom';
import Navbar from './components/Navbar';
import Home from './components/Home';
import About from './components/About';
import Footer from './components/Footer';
import Showlist from './containers/Showlist';
import Showinfo from './containers/Showinfo';

function App() {
  return (
    <Router>
      <div className="App">

        <Navbar />
        <Switch>
          <Route exact path="/">
            <Redirect to="/home" />
          </Route>
          <Route exac path="/home" component={Home} />
          <Route exac path="/about" component={About} />
          <Route exac path="/shows/:id" component={Showinfo} />
          <Route exac path="/shows" component={Showlist} />
        </Switch>
        <Footer />

      </div>
    </Router>
  );
}

export default App;
