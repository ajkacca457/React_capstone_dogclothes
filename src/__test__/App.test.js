import React from 'react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import App from '../App';

describe('App Component', () => {
  it('renders correctly', () => {
    const apptree = TestRenderer
      .create(<App />)
      .toJSON();
    expect(apptree).toMatchSnapshot();
  });
});
