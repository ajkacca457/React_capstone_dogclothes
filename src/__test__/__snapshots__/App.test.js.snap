// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg px-4 position-fixed"
  >
    <img
      alt="tvshowlogo"
      src="navlogo.png"
      style={
        Object {
          "height": "35px",
          "width": "35px",
        }
      }
    />
    <h5
      className="ml-3 text-white"
    >
      TVWORLD
    </h5>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <i
        className="fas fa-bars text-light"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav m-auto text-center"
      >
        <li
          className="nav-item active mr-2"
        >
          <a
            className="nav-link text-white"
            href="/home"
            onClick={[Function]}
          >
            <i
              className="fas fa-home mr-1"
            />
            Home
          </a>
        </li>
        <li
          className="nav-item mr-2"
        >
          <a
            className="nav-link text-white"
            href="/shows"
            onClick={[Function]}
          >
            <i
              className="fas fa-tv mr-1"
            />
            TV shows
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link text-white"
            href="/about"
            onClick={[Function]}
          >
            <i
              className="fas fa-info-circle mr-1"
            />
            About
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="home"
  >
    <div
      className="content d-block d-lg-flex justify-content-between align-items-center"
    >
      <div
        className="leftcont text-white text-center"
      >
        <h2>
          TVWORLD
        </h2>
        <h3
          className="my-2"
        >
          Home of Your Entertainment
        </h3>
        <a
          className="btn btn-warning rounded"
          href="/shows"
          onClick={[Function]}
        >
          Browse Shows
        </a>
      </div>
      <div
        className="vr d-none d-lg-block"
      />
      <div
        className="rightcont d-none d-lg-block"
      >
        <img
          alt="Show1"
          className="tvshow s1"
          src="tvshow1.jpeg"
        />
        <img
          alt="Show1"
          className="tvshow s2"
          src="tvshow2.jpg"
        />
      </div>
    </div>
  </div>
  <div
    className="footer text-white pt-3"
  >
    <p>
      Â© & Project made by Avijit, Microverse
    </p>
  </div>
</div>
`;
